sway/desktop/xwayland.c

```diff
@@ -199,15 +208,32 @@ static void set_fullscreen(struct sway_view *view, bool fullscreen) {
 }
 
 static bool wants_floating(struct sway_view *view) {
-	// TODO:
-	// We want to return true if the window type contains any of these:
-	// NET_WM_WINDOW_TYPE_DIALOG
-	// NET_WM_WINDOW_TYPE_UTILITY
-	// NET_WM_WINDOW_TYPE_TOOLBAR
-	// NET_WM_WINDOW_TYPE_SPLASH
-	//
-	// We also want to return true if the NET_WM_STATE is MODAL.
-	// wlroots doesn't appear to provide all this information at the moment.
+	if (xwayland_view_from_view(view) == NULL) {
+		return false;
+	}
+	struct wlr_xwayland_surface *surface = view->wlr_xwayland_surface;
+	struct sway_xwayland *xwayland = &server.xwayland;
+
+	// TODO: return true if the NET_WM_STATE is MODAL
```

Comment:
What's the reason for not actioning this todo?

Reply:
wlroots doesn't expose this, yet